pygmt/base_plotting.py

```diff
@@ -1918,3 +1918,206 @@ def update_pointers(data_pointers):
+                    (-: option interchanges order) 3: eps1, the most
+                    extensional eigenvalue of strain tensor, with extension
+                    taken positive. **4**: eps2, the most compressional
+                    eigenvalue of strain tensor, with extension taken positive.
+                    **5**: azimuth of eps2 in degrees CW from North.
+
+
+        Other Parameters
+        ----------------
+        vector : bool or str
+            Modify vector parameters. For vector heads, append vector head size
+            [Default is 9p]. For specifying additional attributes, see
+            :gmt-docs:`supplements/geodesy/velo.html#vector-attributes`.
+
+        {B}
+
+        {CPT}
+
+        rescale : str
+            can be used to rescale the uncertainties of velocities
+            (-Se and -Sr) and rotations (-Sw).
+            Can be combined with the confidence variable.
+
+        uncertainty_color : str
+            Sets the color or shade used for filling uncertainty wedges
+            (-Sw) or velocity error ellipses (-Se or -Sr). [If -E is not
+            specified, the uncertainty regions will be transparent].
+            More details on
+            :gmt-docs:`cookbook/features.html#gfill-attrib`.
+
+        facecolor : str
+            Select color or pattern for filling of symbols or polygons
+            [Default is no fill]. More details on
+            :gmt-docs:cookbook/features.html#gfill-attrib`.
+
+        line: str
+            Draw lines. Ellipses and fault planes will have their outlines
+            drawn using current pen (see -W).
+
+        no_clip: str
+            Do NOT skip symbols that fall outside the frame boundary specified
+            by -R. [Default plots symbols inside frame only].
+
+        pen : str
+            Set pen attributes for velocity arrows, ellipse circumference and
+            fault plane edges. [Defaults: width = default, color = black,
+            style = solid].
+
+        {U}
+        {V}
+        {XY}
+        {p}
+        {t}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(data)
+
+        extra_arrays = []
+        if "G" in kwargs and not isinstance(kwargs["G"], str):
+            if kind != "vectors":
+                raise GMTInvalidInput(
+                    "Can't use arrays for color if data is matrix or file."
+                )
+            extra_arrays.append(kwargs["G"])
+            del kwargs["G"]
+
+        if scaling is not None:
+            if not isinstance(scaling,str):
+                raise GMTInvalidInput(
+                    "scaling has to be a string."
+                )
+            extra_arrays.append(scaling)
+
+        with Session() as lib:
+            # Choose how data will be passed in to the module
+            if kind == "file":
+                file_context = dummy_context(data)
+            elif kind == "matrix":
+                file_context = lib.virtualfile_from_matrix(data)
```

Comment:
:warning: Do Not :warning: commit this suggested change yet!! What this does is allow for text label input, **but** it will only work after #684.

```suggestion
            elif kind == "matrix":
                if pd.api.types.is_numeric_dtype(data):
                    file_context = lib.virtualfile_from_matrix(data)
                else:
                    file_context = lib.virtualfile_from_vectors(
                        *[data[column] for column in data]
                    )
```

This is what the example gallery would look like:

![velo plot with text labels](https://user-images.githubusercontent.com/23487320/98432828-0b2ef500-2127-11eb-932f-1f859201baeb.png)